src/Sentry/PlatformAbstractions/FrameworkInfo.NetFx.cs

```diff
@@ -160,13 +186,17 @@ private static FrameworkInstallation GetFromV4(RegistryKey subKey, string subKey
                 },
                 Version = version,
                 ServicePack = subKey.GetInt("SP"),
-                Release = hasRelease ? release : null as int?
+                Release = hasRelease ? release : null
             };
         }
 
         // https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/how-to-determine-which-versions-are-installed#to-find-net-framework-versions-by-querying-the-registry-in-code-net-framework-45-and-later
-        internal static int? Get45PlusLatestInstallationFromRegistry()
+        internal static int? Get45PlusLatestInstallationFromRegistry(bool registryLock)
         {
+            if (registryLock)
+            {
+                return null;
+            }
             using var ndpKey = RegistryKey.OpenBaseKey(RegistryHive.LocalMachine, RegistryView.Registry32)
                 .OpenSubKey(@"SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full\");
```

Comment:
Note: you're not passing a second parameter to `OpenSubKey` here (`bool writable`). I believe that parameter is `true` by default which could cause access exception if the user doesn't have required permissions. Read-only access, on the other hand, should be available in all cases. Of course it still might fail when running on non-Windows operating system.

https://docs.microsoft.com/en-us/dotnet/api/microsoft.win32.registrykey.opensubkey?view=net-5.0#microsoft-win32-registrykey-opensubkey(system-string-system-boolean)

Reply:
After looking at it again, it seems the second parameter is `false` by default, so it probably wasn't the issue.

Reply:
The doc says if you only pass the name it'll try to open as read-only
https://docs.microsoft.com/en-us/dotnet/api/microsoft.win32.registrykey.opensubkey?view=net-5.0#Microsoft_Win32_RegistryKey_OpenSubKey_System_String_

Should I specify the second parameter as false just to be sure?

Reply:
> Should I specify the second parameter as false just to be sure?

I think it's a good idea, yeah